src/twisted/web/http_headers.py

```diff
@@ -209,23 +214,50 @@ def setRawHeaders(self, name, values):
                 "instance of %r instead" % (name, type(values))
             )
 
+        if not isinstance(name, (bytes, str)):
+            raise TypeError(
+                "Header name is an instance of %r, " "not bytes or str" % (type(name),)
+            )
+
+        for count, value in enumerate(values):
+            if not isinstance(value, (bytes, str)):
+                raise TypeError(
+                    "Header value at position %s is an instance of %r, not "
+                    "bytes or str"
+                    % (
+                        count,
+                        type(value),
+                    )
+                )
+
         name = _sanitizeLinearWhitespace(self._encodeName(name))
         encodedValues = [
             _sanitizeLinearWhitespace(v) for v in self._encodeValues(values)
         ]
 
-        self._rawHeaders[name] = self._encodeValues(encodedValues)
+        self._rawHeaders[name] = encodedValues
 
-    def addRawHeader(self, name, value):
+    def addRawHeader(self, name: String, value: String) -> None:
```

Comment:
Here's a place where `AnyStr` would be awkward; `name` and `value` *don't* have to be the same type.  But I'd still rather just inline `Union[bytes, str]` here twice separately.

Reply:
OK, I replaced the private alias `String` with `Union[bytes, str]` everywhere as an go at this